### ISSUE TITLE
Consider @extern classes to be mutable if not annotated with @pure

### ISSUE THREAD

**Issue by romac** [epfl-lara/stainless#268]
Subsumes #267

**Comment by samarion** [comment]
@romac isn't this now fixed?

**Comment by samarion** [comment]
Do we even support `@extern` classes? Don't we only support `@extern` functions?

**Comment by romac** [comment]
Yeah maybe we should disallow it, since we have ignored fields. Or just keep it to mean: mark all methods as extern. WDYT?

**Comment by samarion** [comment]
I like the "mark all methods as extern" meaning. I wonder if it also marks the accessors as extern. That would perhaps be the ideal behavior.

**Comment by samarion** [comment]
Oh also we should probably make sure `@extern` classes are considered as marked with `@mutable` unless they are marked with `@pure` in the sealing phase.

**Comment by romac** [comment]
> I like the "mark all methods as extern" meaning. I wonder if it also marks the accessors as extern. That would perhaps be the ideal behavior.

They are indeed.

**Comment by romac** [comment]
TODO: Merge 11beaa62b94f61c4409a74c67d2aee0c34f7d2d2 on top of `move-imperative` branch.
